source/lz4.c

```diff
@@ -986,6 +986,47 @@ int LZ4_loadDict (LZ4_stream_t* LZ4_dict, const char* dictionary, int dictSize)
     return dict->dictSize;
 }
 
+int LZ4_stream_preserve(LZ4_stream_t* stream_, int** buf_out)
+{
+  int nbytes = sizeof(int) * HASH_SIZE_U32;
+  int* buf = malloc(nbytes);
+  memset(buf, 0, nbytes);
+  *buf_out = buf;
+  LZ4_stream_t_internal* stream = (LZ4_stream_t_internal*)stream_;
+  int buf_pos = 0;
+  for (int i = 0; i < HASH_SIZE_U32; i++)
+  {
+    if (stream->hashTable[i] != 0)
+    {
+      buf[buf_pos] = i;
```

Comment:
@mirw suggested that, rather than being an array of locations, this should be an array of {location, value} structs, as this gets better CPU cache behaviour.